references_test.go

```diff
@@ -18,358 +14,6 @@ type ReferencesSuite struct {
-			"b7015a5d36990d69a054482556127b9c7404a24a",
-			"582da9622e3a72a19cd261a017276d72b5b0051a",
-			"0c5bb1e4392e751f884f3c57de5d4aee72c40031",
-			"c9c2a0ec03968ab17e8b16fdec9661eb1dbea173",
-			"a3cdf880826b4d9af42b93f4a2df29a91ab31d35",
-			"18526c447f5174d33c96aac6d6433318b0e2021c",
-			"2a6288be1c8ea160c443ca3cd0fe826ff2387d37",
-			"9e74d009894d73dd07773ea6b3bdd8323db980f7",
-			"d2f6214b625db706384b378a29cc4c22237db97a",
-			"202a9c720b3ba8106e022a0ad027ebe279040c78",
-			"791bcd1592828d9d5d16e83f3a825fb08b0ba22d",
-			"01e65d67eed8afcb67a6bdf1c962541f62b299c9",
-			"6328ee836affafc1b52127147b5ca07300ac78e6",
-			"3de4f77c105f700f50d9549d32b9a05a01b46c4b",
-			"8980daf661408a3faa1f22c225702a5c1d11d5c9",
-			"8eb116de9128c314ac8a6f5310ca500b8c74f5db",
-			"88e841aad37b71b78a8fb88bc75fe69499d527c7",
-			"370d61cdbc1f3c90db6759f1599ccbabd40ad6c1",
-			"505577dc87d300cf562dc4702a05a5615d90d855",
-			"b5c6053a46993b20d1b91e7b7206bffa54669ad7",
-			"ba486de7c025457963701114c683dcd4708e1dee",
-			"b41d7c0e5b20bbe7c8eb6606731a3ff68f4e3941",
-			"a47d0aaeda421f06df248ad65bd58230766bf118",
-			"495c7118e7cf757aa04eab410b64bfb5b5149ad2",
-			"46670eb6477c353d837dbaba3cf36c5f8b86f037",
-			"dd2d03c19658ff96d371aef00e75e2e54702da0e",
-			"4bbcad219ec55a465fb48ce236cb10ca52d43b1f",
-			"50d0556563599366f29cb286525780004fa5a317",
-			"9a06d3f20eabb254d0a1e2ff7735ef007ccd595e",
-			"d4b48a39aba7d3bd3e8abef2274a95b112d1ae73",
-		}},
-	*/
-	/*
-		{"https://github.com/spinnaker/spinnaker.git", "b32b2aecae2cfca4840dd480f8082da206a538da", "config/default-spinnaker-local.yml", []string{
-			"ae904e8d60228c21c47368f6a10f1cc9ca3aeebf",
-			"99534ecc895fe17a1d562bb3049d4168a04d0865",
-			"caf6d62e8285d4681514dd8027356fb019bc97ff",
-			"eaf7614cad81e8ab5c813dd4821129d0c04ea449",
-			"5a2a845bc08974a36d599a4a4b7e25be833823b0",
-			"41e96c54a478e5d09dd07ed7feb2d8d08d8c7e3c",
-			"974b775a8978b120ff710cac93a21c7387b914c9",
-			"87e459a9a044b3109dfeb943cc82c627b61d84a6",
-			"5e09821cbd7d710405b61cab0a795c2982a71b9c",
-			"8cc2d4bdb0a15aafc7fe02cdcb03ab90c974cafa",
-			"3ce7b902a51bac2f10994f7d1f251b616c975e54",
-			"a596972a661d9a7deca8abd18b52ce1a39516e89",
-			"8980daf661408a3faa1f22c225702a5c1d11d5c9",
-		}},
-	*/
-	/*
-		{"https://github.com/spinnaker/spinnaker.git", "b32b2aecae2cfca4840dd480f8082da206a538da", "config/spinnaker.yml", []string{
-			"ae904e8d60228c21c47368f6a10f1cc9ca3aeebf",
-			"caf6d62e8285d4681514dd8027356fb019bc97ff",
-			"eaf7614cad81e8ab5c813dd4821129d0c04ea449",
-			"5a2a845bc08974a36d599a4a4b7e25be833823b0",
-			"41e96c54a478e5d09dd07ed7feb2d8d08d8c7e3c",
-			"974b775a8978b120ff710cac93a21c7387b914c9",
-			"ed887f6547d7cd2b2d741184a06f97a0a704152b",
-			"d4553dac205023fa77652308af1a2d1cf52138fb",
-			"a596972a661d9a7deca8abd18b52ce1a39516e89",
-			"66ac94f0b4442707fb6f695fbed91d62b3bd9d4a",
-			"079e42e7c979541b6fab7343838f7b9fd4a360cd",
-		}},
-	*/
-}
-
-func (s *ReferencesSuite) TestObjectNotFoundError(c *C) {
```

Comment:
Some of the changes within references feel orthogonal to the PR purpose. Do you mind sharing your thought process on this please?

Reply:
See my comment here: https://github.com/go-git/go-git/pull/780#issuecomment-1580612255

In short, the function `references` was only used by the old implementation of blame, I believe the results that function give are not useful (if not dangerous if used wrong like it was before). It would only really be useful for a file that had no branches in its history. As it is no longer used I removed it (happy to leave it there if you prefer that).

Reply:
You are right, originally `references.go` was [revlist/revlist.go](https://github.com/go-git/go-git/commit/d643cea1e8a6d618b2eddfdbed086c7bdf208658#diff-df6d59375198a3beb203fa89eb592cf8ba5e21ff4b62a6d0b4e61505d51cd87f) but as the code developed it got renamed and a lot of the functionality became blame-centric.